OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -150,14 +150,39 @@ public override Activity Tick(Actor self)
 				return this;
 			}
 
-			if (self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length < aircraft.Info.MinAirborneAltitude)
-				QueueChild(self, new TakeOff(self), true);
+			var delta = attackAircraft.GetTargetPosition(pos, checkTarget) - pos;
+			var desiredFacing = delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : aircraft.Facing;
+			var isAirborne = self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length >= aircraft.Info.MinAirborneAltitude;
+
+			if (!aircraft.Info.CanHover)
+			{
+				if (!isAirborne)
+					QueueChild(self, new TakeOff(self), true);
 
-			if (attackAircraft != null && target.IsInRange(self.CenterPosition, attackAircraft.GetMinimumRange()))
-				QueueChild(self, new FlyTimed(ticksUntilTurn, self), true);
+				if (attackAircraft != null && target.IsInRange(self.CenterPosition, attackAircraft.GetMinimumRange()))
+					QueueChild(self, new FlyTimed(ticksUntilTurn, self), true);
 
-			QueueChild(self, new Fly(self, target, checkTarget.CenterPosition, Color.Red), true);
-			QueueChild(self, new FlyTimed(ticksUntilTurn, self));
+				QueueChild(self, new Fly(self, target, checkTarget.CenterPosition, Color.Red), true);
+				QueueChild(self, new FlyTimed(ticksUntilTurn, self));
+			}
+			else if (!isAirborne)
+				QueueChild(self, new TakeOff(self), true);
```

Comment:
If !CanHover we queue the takeoff and the activities to do after takeoff in one go, whereas CanHover queus just the takeoff and waits until this next ticks to queue the movement.

We should be able to reduce duplication by putting
```csharp
if (!isAirborne)
	QueueChild(self, new TakeOff(self), true);
```

before the `if (!aircraft.Info.CanHover)` and letting both cases queue their movements after it?